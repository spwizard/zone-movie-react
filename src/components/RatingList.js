import React, { Component } from "react";
import { connect } from "react-redux";
import { Col, FormGroup, Input, Label, Row } from 'reactstrap';
import { selectMoviesByRating } from '../actions/user';

class RatingList extends Component {

  handleSliderChange = (evt) => {
    this.props.selectRating(evt.target.value)
  }

  render() {
    return (
      <Row style={{
        marginBottom: 20
      }}>
        <Col sm="12"><h4>Rating selector ({this.props.rating})</h4></Col>
        <Col sm="12">
        <input 
          type="range" 
          onChange={this.handleSliderChange}
          min="0"
          max="10"
          value={this.props.rating}
          step="0.5" />
        </Col>
      </Row>
    );
  }
}

const mapStateToProps = state => {
  return {
    rating: state.user.selectedRating
  }
};

const mapDispatchToProps = (dispatch) => ({
  selectRating: (id) => dispatch(selectMoviesByRating(id))
});


export default connect(
  mapStateToProps,
  mapDispatchToProps
)(RatingList)