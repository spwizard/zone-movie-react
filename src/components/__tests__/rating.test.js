import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import configureStore from "redux-mock-store"; // Smart components

// Component to be tested
import RatingList from "../RatingList";

const mockStore = configureStore();
const initialState = {
    user : {
        selectedGenres: [],
        selectedRating: 3
    }
};
const store = mockStore(initialState);

describe("<RatingList />", () => {
    test('renders the component', () => {
        const wrapper = shallow(<RatingList store={store} />);
    const component = wrapper.dive();
  
        expect(toJson(component)).toMatchSnapshot();
      });
});
