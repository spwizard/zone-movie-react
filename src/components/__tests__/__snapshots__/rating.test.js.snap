// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RatingList /> renders the component 1`] = `
<Row
  style={
    Object {
      "marginBottom": 20,
    }
  }
  tag="div"
>
  <Col
    sm="12"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <h4>
      Rating selector (
      3
      )
    </h4>
  </Col>
  <Col
    sm="12"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <input
      max="10"
      min="0"
      onChange={[Function]}
      step="0.5"
      type="range"
      value={3}
    />
  </Col>
</Row>
`;
